src/realtimecontroller/DefaultRealtimeController.ts

```diff
@@ -413,13 +414,12 @@ export default class DefaultRealtimeController implements RealtimeController {
 
   realtimeReceiveDataMessage(dataMessage: DataMessage): void {
     try {
-      if (this.state.receiveDataMessageCallbacks.has(dataMessage.topic)) {
-        for (const fn of this.state.receiveDataMessageCallbacks.get(dataMessage.topic)) {
-          fn(dataMessage);
-        }
-      }
+      iterateEvery(this.state.receiveDataMessageCallbacks.get(dataMessage.topic), fn => {
+        fn(dataMessage);
+      });
     } catch (e) {
-      this.onError(e);
+      // We don't want to throw to our caller, but we still want to surface the error in the console
+      Promise.reject(e);
```

Comment:
What is the issue you ran into? I think it is intentional in any real time API here to throw a hard error.

Reply:
There was a parsing issue in the app code handling transcription that threw an exception. The issue is that the audio video controller listens to fatals and shuts down the meeting, even though something as simple as a parsing bug really shouldn't be considered fatal